filters/include/pcl/filters/impl/voxel_grid.hpp

```diff
@@ -339,7 +345,9 @@ pcl::VoxelGrid<PointT>::applyFilter (PointCloud &output)
 
   // Second pass: sort the index_vector vector using value representing target cell as index
   // in effect all points belonging to the same output cell will be next to each other
-  std::sort (index_vector.begin (), index_vector.end (), std::less<cloud_point_index_idx> ());
+  //std::sort (index_vector.begin (), index_vector.end (), std::less<cloud_point_index_idx> ());
+  boost::sort::spreadsort::integer_sort(index_vector.begin (), index_vector.end (),
+                                        cloud_point_index_rightshift (), std::less<cloud_point_index_idx> ());
```

Comment:
Is it possible to provide a lambda here?

Reply:
That would be nicer. I will check that out